src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseSubmitter.java

```diff
@@ -108,43 +112,53 @@ public String submit(String currentPath)
      * Submits folder of exercise to TMC. Finds it from current directory.
      *
      * @param currentPath path from which this was called.
-     * @param observer {@link ProgressObserver} that is informed of the submission progress
-     * @return String with url from which to get results or null if exercise was not found.
+     * @param observer {@link ProgressObserver} that is informed of the
+     * submission progress
+     * @return URL from which to get results or null if exercise was
+     * not found.
      * @throws IOException if zip creation fails
```

Comment:
Consider leaving throws-clauses away from javadoc. Google styleguide http://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s7-javadoc does not give a best practise for them but I think they make code less readable.